src/component.ts

```diff
@@ -69,7 +77,8 @@ export function componentFactory (
     : Vue
   const Extended = Super.extend(options)
 
-  forwardStaticMembers(Extended, Component, Super)
+  forwardStaticMembersAndCollectReflection(Extended, Component, Super, reflectionMap)
+  copyReflectionMetadata(Component, Extended, reflectionMap)
```

Comment:
I think we don't need this if we directly forward metadata and remove `reflectionMap`.